xgi/linalg/hypergraph_matrix.py

```diff
@@ -294,3 +296,56 @@ def clique_motif_matrix(H, sparse=True, index=False):
     W, rowdict = adjacency_matrix(H, sparse=sparse, weighted=True, index=True)
 
     return (W, rowdict) if index else W
+
+
+def adjacency_tensor(H, order, index=False):
+    """
+    Compute the order-d adjacency tensor of a hypergraph from its incidence matrix.
+
+    The order-d adjacency tensor B is defined as B[i1, i2, ..., id] = 1 if there exists
+    a hyperedge containing nodes (i1, i2, ..., id), and 0 otherwise.
+
+    Parameters
+    ----------
+    H : xgi.Hypergraph
+        The input hypergraph.
+    order : int
+        The order of interactions to consider.
+    index: bool, default: False
+        Specifies whether to output disctionaries mapping the node IDs to indices
+
+    Returns
+    -------
+    B : np.ndarray
+        Adjacency tensor
+    """
+
+    I, rowdict, _ = incidence_matrix(H, order=order, sparse=False, index=True)
+
+    if I.shape == (0, 0):
+        if not rowdict:
+            B = np.empty((0,) * (order + 1))
+        if not coldict:
+            shape = (H.num_nodes,) * (order + 1)
+            B = np.zeros(shape, dtype=int)
+        return (B, {}) if index else B
+
+    # find nodes participating in each hyperedge
+    hyperedges = H.edges.filterby("order", order).members()
+
+    map_node2index = {v: k for (k, v) in rowdict.items()}
+    hyperedges_node_ids = [
```

Comment:
Is this for performance? Why do you need to construct this list of lists when you can just do this for each edge? I.e., `idx = [nodedict[n] for n in e]` inside the for loop. I feel like that would use less memory.

Reply:
Good point, done